web/react/utils/text_formatting.jsx

```diff
@@ -123,27 +124,60 @@ function autolinkUrls(text, tokens) {
 }
 
 function autolinkAtMentions(text, tokens) {
-    let output = text;
+    // Return true if provided character is punctuation
+    function isPunctuation(character) {
+        const re = /[\u2000-\u206F\u2E00-\u2E7F\\'!"#$%&()*+,\-.\/:;<=>?@\[\]^_`{|}~]/g;
```

Comment:
Heh, fair enough. How bizarre. And it seemed to fit the issue you were having too! Drats.